mongorestore/restore.go

```diff
@@ -301,6 +332,89 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
+func convertLegacyIndexes(indexes []IndexDocument) {
+	for _, index := range indexes {
+		convertLegacyIndexKeys(index)
+		convertLegacyIndexOptions(index)
+	}
+}
+
+func convertLegacyIndexKeys(index IndexDocument) {
+	var converted bool
+	originalJSONString := createExtJSONString(index.Key)
+	for j, elem := range index.Key {
+		switch v := elem.Value.(type) {
+		case int32, int64, float64:
+			// Only convert 0 value
+			if v == 0 {
+				index.Key[j].Value = 1
+				converted = true
+			}
+		case primitive.Decimal128:
+			if isZero(v) {
+				index.Key[j].Value = 1
+				converted = true
+			}
+		case string:
+			// Only convert an empty string
+			if v == "" {
+				index.Key[j].Value = 1
+				converted = true
+			}
+		default:
+			// Convert all types that aren't strings or numbers
+			index.Key[j].Value = 1
+			converted = true
+		}
+	}
+	if converted {
+		newJSONString := createExtJSONString(index.Key)
+		log.Logvf(log.Always, "convertLegacyIndexes: converted index values '%s' to '%s' on collection '%s'",
+			originalJSONString, newJSONString, index.Options["ns"])
+	}
+}
+
+func isZero(num primitive.Decimal128) bool {
+	h, l := num.GetBytes()
+	if l != 0 {
+		return false
+	}
+	for i := uint64(0); i < 49; i++ {
+		if h&(1<<i) != 0 {
+			return false
+		}
+	}
+	return true
+}
+
+func convertLegacyIndexOptions(index IndexDocument) {
+	var converted bool
+	originalJSONString := createExtJSONString(index.Options)
+	for key := range index.Options {
+		if _, ok := validIndexOptions[key]; !ok {
+			delete(index.Options, key)
+			converted = true
+		}
+	}
+	if converted {
+		newJSONString := createExtJSONString(index.Options)
+		log.Logvf(log.Always, "convertLegacyIndexes: converted index options '%s' to '%s'",
+			originalJSONString, newJSONString)
+	}
+}
+
+func createExtJSONString(doc interface{}) string {
+	// by default return "<unable to format document>"" since we don't
```

Comment:
I made a call here not to throw an error when formatting informational messages. Let me know if you think we should be more strict.